system/Test/Mock/MockCache.php

```diff
@@ -284,6 +273,69 @@ public function isSupported(): bool
 		return true;
 	}
 
+	//--------------------------------------------------------------------
+	// Test Helpers
 	//--------------------------------------------------------------------
 
+	/**
+	 * Instructs the class to ignore all
+	 * requests to cache an item, and always "miss"
+	 * when checked for existing data.
+	 *
+	 * @return $this
+	 */
+	public function bypass(bool $bypass = true)
+	{
+		$this->clean();
+
+		$this->bypass = $bypass;
+
+		return $this;
+	}
+
+	//--------------------------------------------------------------------
+	// Additional Assertions
+	//--------------------------------------------------------------------
+
+	/**
+	 * Asserts that the cache has an item named $key.
+	 * The value is not checked since storing false or null
+	 * values is valid.
+	 *
+	 * @param string $key
+	 */
+	public function assertHas(string $key)
+	{
+		Assert::assertNotNull($this->get($key), "The cache does not have an item named: `{$key}`");
+	}
+
+	/**
+	 * Asserts that the cache has an item named $key with a value matching $value.
+	 *
+	 * @param string $key
+	 * @param null   $value
+	 */
+	public function assertHasValue(string $key, $value = null)
+	{
+		$item = $this->get($key);
+
+		// Let assertHas handle throwing the error for consistency
+		// if the key is not found
+		if (empty($item))
+		{
+			$this->assertHas($key);
+		}
+
+		Assert::assertEquals($value, $this->get($key), "The cached item `{$key}` does not equal match expectation. Found: " . print_r($value, true));
+	}
+
+	/**
+	 * Asserts that the cache does NOT have an item named $key.
+	 *
+	 * @param string $key
+	 */
+	public function assertMissing(string $key)
+	{
+		Assert::assertFalse(array_key_exists($key, $this->cache), "The cached item named `{$key}` exists.");
```

Comment:
You can use `Assert::arrayNotHasKey` for this.

Reply:
Good call. My knowledge of the available assertions is definitely fairly simple. I've modified it with your suggestion.